torch/_inductor/ir.py

```diff
@@ -2677,30 +2677,38 @@ def apply_constraint(self):
 
     def codegen_args(self):
         args = [x.codegen_reference() for x in self.inputs]
-        args.extend(map(repr, self.constant_args))
-        return args
-
-    def cpp_wrapper_codegen_args(self):
-        args = [x.codegen_reference() for x in self.inputs]
-        args.extend(self.cpp_constant_args)
+        args.extend(
+            map(
+                V.graph.wrapper_code.val_to_str,
+                self.constant_args,
+            )
+        )
         return args
 
     def codegen_kwargs(self):
         kwargs = []
         if self.kwargs:
             if V.graph.cpp_wrapper:
-                for arg_name in self.ordered_kwargs_for_cpp_kernel:
-                    assert arg_name in self.kwargs, (
-                        "arg %s not found in self.kwargs" % arg_name
-                    )
-                    v = self.kwargs.get(arg_name)
-                    kwargs.append(repr(v))
+                if self.kwargs:
+                    # TODO: use native_functions.yaml as the ground truth
+                    assert (
```

Comment:
This should partly address your comment at https://github.com/pytorch/pytorch/pull/98741#discussion_r1162498456